src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs

```diff
@@ -400,6 +412,125 @@ public DocumentConflict ResolveToLatest(List<DocumentConflict> conflicts)
+                return;
+
+            var currentDocumentAttachments = (from BlittableJsonReaderObject bjro in attachments select JsonDeserializationClient.AttachmentName(bjro)).ToList();
+            var storageAttachmentsDetails = _database.DocumentsStorage.AttachmentsStorage.GetAttachmentDetailsForDocument(context, lowerId);
+
+            var oldAttachmentsMetadata = ExtractOldAttachmentsMetdataIfNeeded(attachments.Length, new List<BlittableJsonReaderObject> { local.Data, conflict.Doc });
+            var attachmentsToRemove = oldAttachmentsMetadata?.Select(x => x.Name).ToList();
+
+            foreach (var currentAttachment in currentDocumentAttachments)
+            {
+                var count = CountInAttachmentsList(currentAttachment.Name, storageAttachmentsDetails);
+                switch (count)
+                {
+                    case 1:
+                        RemoveFromAttachmentsToRemoveListIfNeeded(currentAttachment.Name);
+                        continue;
+                    case 0:
+                        // attachment exists in metadata, does not exist in storage (renamed/added in conflict resolution)
+                        // try to find attachments in attachment storage with both same hash and content type
+                        var matches = storageAttachmentsDetails.Where(x => x.Hash == currentAttachment.Hash && x.ContentType == currentAttachment.ContentType).ToList();
+                        if (matches.Count == 0)
+                        {
+                            // attachment is in the metadata but doesn't exist in the attachments storage
+                            throw new AttachmentConflictException($"Attachment '{currentAttachment.Name}' was found in '{nameof(DocumentFlags.Resolved)}' document '{lowerId}' metadata, but was not found in the attachments storage using the following: Hash: '{currentAttachment.Hash}', ContentType: '{currentAttachment.ContentType}'.", lowerId.ToString());
+                        }
+
+                        foreach (var match in matches)
+                        {
+                            // find which name of the matched attachments doesn't exist in the current metadata
+                            if (CountInAttachmentsList(match.Name, currentDocumentAttachments) == 0)
+                            {
+                                if (_database.DocumentsStorage.AttachmentsStorage.RenameAttachment(context, lowerId, match.Name, match.Hash, match.ContentType, AttachmentType.Document, currentAttachment.Name) == null)
+                                    throw new AttachmentConflictException($"Could not rename attachment '{match.Name}' to '{currentAttachment.Name}' in attachments storage of document '{lowerId}', attachment has the following: Hash: '{currentAttachment.Hash}', ContentType: '{currentAttachment.ContentType}'.", lowerId.ToString());
```

Comment:
Can use `_1`, `_2`, etc instead.
Can use random value, etc.

We can't throw here.

Reply:
`RenameAttachment` should not fail, its a precaution so it can be resolved manually if something goes wrong and we cannot add 1,2, random value, because this is result of scripts of the `ScriptResolver`